programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/adapter/VFSFileObjectAdapter.java

```diff
@@ -597,33 +586,42 @@ public DataSpacesFileObject ensureExistingOrSwitch(boolean checkWriteProtected)
         return null;
     }
 
-    private static boolean checkIfFileWritable(DataSpacesFileObject dsfo) throws FileSystemException {
-        if (!dsfo.isWritable()) {
+    private static boolean checkIfFileWritable(DataSpacesFileObject fileObject) throws FileSystemException {
```

Comment:
Can't the write just fail, why checking for it? The code which deals with the write must handle the writes and their failures. We could write less code and code which handles failures at a higher level is actually rewarded and failures are not hidden.